src/Window.cpp

```diff
@@ -1432,33 +1695,205 @@ bool Window::moveForward(int distance) {
     int moves = 1;
     while (moves < distance) {
         if (wallFront(moves)) {
-            break;
+            stopRun("Stop running because of a wall!");
+            return false;
         }
         moves += 1;
     }
-    m_movement = Movement::MOVE_FORWARD;
+    m_movement = Movement::MOVE_FULLFORWARD;
     m_doomedToCrash = (moves != distance);
     m_movesRemaining = moves;
-    if (m_startingLocation.first == 0 && m_startingLocation.second == 0) {
+
+    if (m_startingLocation.first == 1 && m_startingLocation.second == 1) {
         stats->startRun();
     }
     // increase the stats by the distance that will be travelled
-    stats->addDistance(moves);
+
+    if(real)
+    {
+    switch(distance) {
+    case 1:
+        m_speedSlider->setValue(m_sliderValue);
+        break;
+    case 2 :
+        m_speedSlider->setValue(m_sliderValue*1.25);
+        break;
+    case 3:
+        m_speedSlider->setValue(m_sliderValue*1.5);
+        break;
+    default :
+        m_speedSlider->setValue(m_sliderValue*2.0);
```

Comment:
Naively, doesn't make the slider value a function of the current value cause weird behavior? Where the slider increases to the maximum gradually, or vice versa? Is that the actual correct behavior? Or should we have something smarter where if the user specifies that the mouse should move N spaces forward, we can generate the "ideal" acceleration curve for the robot and then use that the guide the movement.

Reply:
currently the logic just simulates maximum acceleration and distances>2 but yeah, you are right, i should map the value of the acceleration with the ration of N. so that the more N the more acceleration